src/main/java/net/glowstone/block/itemtype/ItemKnowledgeBook.java

```diff
@@ -19,6 +41,8 @@ public void rightClickAir(GlowPlayer player, ItemStack holding) {
                         .getRecipeByKey(recipe), true);
                 }
             }
+
+            holding.setAmount(0);
```

Comment:
It should also set the type to air.

Reply:
Both handlers create an empty stack if necessary ([example](https://github.com/GlowstoneMC/Glowstone/pull/644/files#diff-7544280850f3676c540cf78ab99c1db1R97)) after executing `rightClickAir` or `rightClickBlock`.

Reply:
Then why did knowledge books stay in my inventory after use?